WEB/Src/WindowsServer/WindowsServer/Implementation/AzureComputeMetadataHeartbeatPropertyProvider.cs

```diff
@@ -101,5 +103,25 @@ public async Task<bool> SetDefaultPayloadAsync(IHeartbeatPropertyManager provide
 
             return hasSetFields;
         }
+
+        public void Dispose()
```

Comment:
Who will call this method?

Reply:
The call stack here is: TelemetryModule > Provider > Requestor > HttpClient.

Inside the TelemetryModule, I moved the Provider into a using to ensure it gets disposed.

Reply:
Stepping back, if the whole purpose is to allow mock in the unit test, would reflection (setting some private member instead of having to change any production code) work?

Reply:
I'm not sure that a private member helps us here.

The original class doesn't have a private HttpClient. It creates the client just-in-time to make a request.
This is how it avoided the responsibility of Disposing.
https://github.com/microsoft/ApplicationInsights-dotnet/blob/a90eb8fa1013b5aaf484bc740b97d22a2df8141d/WEB/Src/WindowsServer/WindowsServer/Implementation/AzureMetadataRequestor.cs#L88-L110



I can mock any HTTP response if I can provide a custom HttpMessageHandler into the HttpClient.
```
var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
new HttpClient(mockHttpMessageHandler.Object)
```

Reply:
I came up with an alternate approach that works.
Tomorrow i'll clean it up and create a new PR.